src/content-tags-drawer/data/apiHooks.jsx

```diff
@@ -112,11 +112,12 @@ export const useContentTaxonomyTagsData = (contentId) => (
 /**
  * Builds the query to get meta data about the content object
  * @param {string} contentId The id of the content object (unit/component)
+ * @param {boolean} enabled Flag to enable/disable the query
  */
-export const useContentData = (contentId) => (
+export const useContentData = (contentId, enabled) => (
   useQuery({
     queryKey: ['contentData', contentId],
-    queryFn: () => getContentData(contentId),
+    queryFn: enabled ? () => getContentData(contentId) : undefined,
```

Comment:
I don't think we should enable/disable the whole api hook just because sometimes we don't want to fetch this data.. Is this a pattern we use elsewhere?

Reply:
We only need this hook to get the item title when we use the `ContentTagsDrawer` on the Course/Unit outline (`variant: 'drawer'`)
![image](https://github.com/user-attachments/assets/17ea67fc-5b2a-451c-807b-faf1691a6406)

For libraries, it was unnecessarily called for components, and returning null for collections:
https://github.com/openedx/frontend-app-authoring/blob/451b821c3b5a8b25494c2315ab4f4e6aacd56ee4/src/content-tags-drawer/data/api.js#L79-L83

Since we shouldn't conditionally call hooks (https://react.dev/reference/rules/rules-of-hooks#only-call-hooks-at-the-top-level), I think it is better to disable it than move this rule to the API and return `null`.

Would you mind sharing your concerns here?

Reply:
@rpenido 

> Since we shouldn't conditionally call hooks

Ahh.. you're totally right, I forgot about that. Please disregard my comment!